### PR TITLE
Count new entities created by an element marking

### PR DESCRIPTION
**Pull request by bgranzow**

### CODE REVIEW

File: src/Omega_h_amr_topology.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+void mark_amr(Mesh* mesh, Read<Byte> elem_mark) {
+  auto dim = mesh->dim();
+  for (Int i = 1; i <= dim; ++i) {
+    auto dim_mark = mark_down(mesh, dim, i, elem_mark);
+    mesh->add_tag<Omega_h::Byte>(i, "refine", 1, dim_mark);
+  }
+}
+
+Few<Real, 4> count_amr(Mesh* mesh) {
```
Comment (ibaned) [inline src/Omega_h_amr_topology.cpp]: This needs to be `Few<LO, 4>`, entity counts are not floating-point values.

File: src/Omega_h_amr_topology.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  auto dim = mesh->dim();
+  Few<Real, 4> num_ents({0,0,0,0});
+  for (Int i = 1; i <=dim; ++i) {
+    auto dim_tag = mesh->get_tag<Byte>(i, "refine");
```
Comment (ibaned) [inline src/Omega_h_amr_topology.cpp]: There is a convenience function for the array of a tag:
```cpp
auto dim_marks = mesh->get_array<Byte>(i, "refine");
```

File: src/Omega_h_hypercube.hpp [hunk 3]
```diff
@@ -485,6 +488,10 @@ OMEGA_H_INLINE constexpr Int hypercube_split_degree(
   return hypercube_degree(parent_dim, parent_dim - split_dim);
 }
 
+void mark_amr(Mesh* mesh, Read<Byte> elem_mark);
```
Comment (ibaned) [inline src/Omega_h_hypercube.hpp]: There should be a new file `Omega_h_amr_topology.hpp` (or something with AMR in the name), this stuff doesn't belong in the hypercube header.

**Comment by bgranzow** [comment]
Hopefully corrected those requested changes...

**Pull request merged**